hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/utils/DeploymentHelper.java

```diff
@@ -102,12 +104,19 @@ public static void successCancellation(final JpaAction action, final ActionRepos
      * @return the result of the action
      */
     public static <T> T runInNewTransaction(@NotNull final PlatformTransactionManager txManager,
-            final String transactionName, final int isolationLevel, @NotNull final TransactionCallback<T> action) {
+            final String transactionName, final int isolationLevel, @NotNull final TransactionCallback<T> action) throws
+            TransactionExecutionException {
         final DefaultTransactionDefinition def = new DefaultTransactionDefinition();
         def.setName(transactionName);
         def.setReadOnly(false);
         def.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
         def.setIsolationLevel(isolationLevel);
-        return new TransactionTemplate(txManager, def).execute(action);
+        final TransactionTemplate transactionTemplate = new TransactionTemplate(txManager, def);
+
+        try {
+            return transactionTemplate.execute(action);
+        } catch (final Exception e) {
```

Comment:
Does it make sense to rethrow the TransactionExecutionException here? I would rather catch the TransactionException and Runtime exception and just log the error with the transaction name here. The reason behind my idea is that I couldn't find any valuable use (e.g. retrying a transaction) of catching the TransactionExecutionException except of just logging. What do you think?

Reply:
Yes, that's a good idea